### PR TITLE
Add current theme url to site meta

### PR DESCRIPTION
**Pull request by slafleche**
This PR adds to the gdn object the current theme's url path. This can be used to import assets from the theme with JS.

### CODE REVIEW

**Comment by tburry** [comment]
We originally used editor.css in the editor and that was removed. Did you find the cause for this? Creating another css file without knowing what happened to the first bit of functionality is something we should try and avoid.

**Comment by charrondev** [comment]
That was my primary concern with https://github.com/vanilla/vanilla/pull/9110 as well.

The line is commented out, but why was it removed in the first place?

**Comment by slafleche** [comment]
So, the code seems to have been commented since the first commit in 2015:

https://github.com/vanilla/vanilla/commit/51207e75f5a9c1fe8cdbd028548d1214055179a3#diff-4cdf1a850be5a6b9c69af45f60dbf08b

There IS and editor CSS file, but it lives in the plugin. We don't want to overwrite that, because we'll need to rewrite all the styles. I think a separate sheet is the best option here @charrondev

File: library/core/class.controller.php [hunk 1]
```diff
@@ -287,6 +287,10 @@ public function __construct() {
 
         parent::__construct();
         $this->ControllerName = strtolower($this->ClassName);
+
+        // Add current theme path to GDN object
+        $themeInfo = Gdn::themeManager()->getThemeInfo(Gdn::themeManager()->currentTheme());
```
Comment (charrondev) [inline library/core/class.controller.php]: The theme manager is deprecated.

The easiest way to get this now is likely with:

```php
/** Addon $currentTheme */
$currentTheme = Gdn::getContainer()->get(ContainerUtils::currentTheme());
```

File: library/core/class.controller.php [hunk 2]
```diff
@@ -12,6 +12,7 @@
  * @abstract
  */
 
+use Vanilla\Utility\ContainerUtils as ContainerUtilsAlias;
```
Comment (initvector) [inline library/core/class.controller.php]: Unused, so I think we can dump it.

```suggestion
```

File: plugins/editor/js/editor.js [hunk 3]
```diff
@@ -47,6 +39,9 @@
             stylesheetsInclude.push(editorWysiwygCSS + '?v=' + editorVersion);
         }
 
+        // If you want to include a custom stylesheet inside the iFrame of the wysiwyg
+        stylesheetsInclude.push(gdn.meta.currentThemePath + '/design/editorWysiwyg.css?v=' + editorVersion);
```
Comment (initvector) [inline plugins/editor/js/editor.js]:
```suggestion
        var currentThemePath = gdn.definition("currentThemePath", null);
        if (currentThemePath) {
            stylesheetsInclude.push(currentThemePath + "/design/editorWysiwyg.css?v=" + editorVersion);
        }
```

Even still, isn't this just gonna throw a 404 in the background and cause a pointless request to the server if the theme doesn't have an editorWysiwyg.css?

**Comment by initvector** [comment]
It looks like Advanced Editor [already mostly supported this](https://github.com/vanilla/vanilla/blob/c246ed34c1f20883d4066fe20d45c98aca20c55c/plugins/editor/class.editor.plugin.php#L545) by adding the right file to your theme.

**Comment by slafleche** [comment]
I was afraid overwriting it would remove important styles, but it looks like all the styles in the wysiwyg file also exist in editor.css... so they could of just left the file blank, but they didn't.

I'm going to leave the theme path url because I think it's pretty handy for theming.

**Pull request merged**